src/csvcubed/utils/sparql_handler/csvw_inspector.py

```diff
@@ -6,17 +6,23 @@
 import rdflib
 
 from csvcubed.models.csvwtype import CSVWType
-from csvcubed.models.sparqlresults import CatalogMetadataResult
+from csvcubed.models.sparqlresults import (
+    CatalogMetadataResult,
+    PrimaryKeyColNamesByDatasetUrlResult,
+    TableSchemaPropertiesResult,
+)
 from csvcubed.utils.sparql_handler.sparql import path_to_file_uri_for_rdflib
 from csvcubed.utils.sparql_handler.sparqlquerymanager import (
     ask_is_csvw_code_list,
     ask_is_csvw_qb_dataset,
     select_csvw_catalog_metadata,
+    select_primary_key_col_names_by_csv_url,
+    select_table_schema_properties,
 )
 
 
 @dataclass
-class CsvWState:
+class CsvWInspector:
     rdf_graph: rdflib.ConjunctiveGraph
     csvw_json_path: Path
 
```

Comment:
Should all cached properties be private? Both catalog_metadata and csvw_type are currently not set as private.

Reply:
Not necessarily. If they're to be directly accessed by a user outside the class they should be public. Since `csvw_type` is used directly (and not via a `get_` function - since this function would do nothing useful) by external callers (and not just tests), it should be public. 

The same will be true of `catalog_metadata` once we've written functions which map the cube/code-list URIs to the catalogue_metadata associated with them.